tracing/src/main/java/com/palantir/tracing/Trace.java

```diff
@@ -28,62 +33,209 @@
  * Represents a trace as an ordered list of non-completed spans. Supports adding and removing of spans. This class is
  * not thread-safe and is intended to be used in a thread-local context.
  */
-public final class Trace {
+public abstract class Trace {
```

Comment:
Not 100% sure, but I think this might be an ABI break.

A quick search internally suggested that nobody is actually importing the `com.palantir.tracing.Trace` class - perhaps we could even make it package-private

Reply:
It’s exposed by Tracer.getAndClearTrace, however all methods are package private — it is only available as a marker